### ISSUE TITLE
kitty send-text workflow broken by 0.32

### ISSUE THREAD

**Issue by xxyxxyxyx1** [kovidgoyal/kitty#7074]
**Describe the bug**
I have used [vim-slime](https://github.com/jpalardy/vim-slime/) with kitty successfully with kitty versions 0.26 – 0.31. However, on kitty v0.32 and v0.32.1, I have keyboard problems. (Reinstalling v0.31 fixes the problems.) 

vim-slime uses [a short piece of vimscript](https://github.com/jpalardy/vim-slime/blob/d9e81b40cd78d8bf7e0c4542fe7182eb802b4ecd/autoload/slime/targets/kitty.vim) to send text from vim to another kitty window, such as an ipython interpreter. 

After the first text is sent with vim-slime, most keyboard shortcuts break, along with most uses of the shift key. For example:
- I cannot type symbols requiring shift in vim, such as `_`; I just get `-`. I can, however, type capital letters.
- kitty keyboard shortcuts do not work, such as `ctrl-tab` to switch tabs
- OS shortcuts do not work while kitty is focused, for example `cmd-q` does not quit kitty. I must use the mouse to quit kitty. 

Functionality is restored only after restarting kitty. The vim-slime seems to work as intended, sending text over to the other kitty window as usual. 

I am not sure where to start looking for this problem. I posted [an issue with vim-slime](https://github.com/jpalardy/vim-slime/issues/412) and was recommended to post here. Any suggestions for tracking down the problem would be appreciated. 

I have tried some small manual tests with kitty, to see if I can break it without using vim-slime, but I have not been successful. I tried for example `kitty @ --to unix:/tmp/mykitty-1975 send-text --match id:2 --stdin ls`, and no keyboard problems occurred.

**To Reproduce**
1. open vim or neovim with the vim-slime plugin
2. send something over slime using a command, `<C-c><C-c>`, or a different remapped shortcut
3. See error

**Screenshots**

**Environment details**
- current vim-slime
- kitty v0.26–0.31, versus kitty v0.32 -v0.32.1
- macOS 14.2.1
- en-US keyboard and languages
- kitty debug output:
```
kitty 0.32.1 (08d88af2fb) created by Kovid Goyal
Darwin mccec1150.local 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:30:44 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6000 arm64
ProductName:		macOS ProductVersion:		14.3 BuildVersion:		23D56
Frozen: True
Paths:
  kitty: /Applications/kitty.app/Contents/MacOS/kitty
  base dir: /Applications/kitty.app/Contents/Resources/kitty
  extensions dir: /Applications/kitty.app/Contents/Resources/Python/lib/kitty-extensions
  system shell: /bin/zsh
Loaded config files:
  test.conf

Config options different from defaults:
allow_remote_control yes
listen_on            unix:/tmp/mykitty
Removed shortcuts:
	cmd+, →  edit_config_file
	cmd+enter →  new_window
	cmd+h →  hide_macos_app
	cmd+k →  clear_terminal to_cursor active
	cmd+m →  minimize_macos_window
	cmd+n →  new_os_window
	cmd+q →  quit
	cmd+t →  new_tab
	cmd+w →  close_tab
	ctrl+cmd+, →  load_config_file
	ctrl+cmd+f →  toggle_fullscreen
	opt+cmd+h →  hide_macos_other_apps
	opt+cmd+r →  clear_terminal reset active
	opt+cmd+s →  toggle_macos_secure_keyboard_entry
	shift+cmd+/ →  open_url https://sw.kovidgoyal.net/kitty/
	shift+cmd+[ →  previous_tab
	shift+cmd+] →  next_tab
	shift+cmd+d →  close_window
	shift+cmd+w →  close_os_window

Important environment variables seen by the kitty process:
	PATH                                /Applications/kitty.app/Contents/MacOS:/Users/redacted/bin/syncscripts:/usr/local/lib/ruby/gems/2.6.0/bin:/usr/local/opt/ruby/bin:/usr/local/opt/gnu-sed/libexec/gnubin:/usr/local/opt/coreutils/libexec/gnubin:/Users/redacted/bin:/opt/homebrew/Caskroom/miniforge/base/envs/npaccel/bin:/opt/homebrew/Caskroom/miniforge/base/condabin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/X11/bin:/Library/Apple/usr/bin:/usr/local/MacGPG2/bin:/Library/TeX/texbin:/Users/redacted/.cargo/bin
	LANG                                en_US.UTF-8
	EDITOR                              nvim
	SHELL                               /bin/zsh
	DISPLAY                             /private/tmp/com.apple.launchd.jXcBahEJMY/org.xquartz:0
	USER                                redacted
	LC_ALL                              en_US.UTF-8
	LC_TERMINAL_VERSION                 3.4.21
	LC_CTYPE                            en_US.UTF-8
	LC_TERMINAL                         iTerm2

```

**Additional context**

I am starting kitty with `kitty --config test.conf` and this minimal test.conf:
```
allow_remote_control yes
listen_on unix:/tmp/mykitty
```

I am starting neovim with `neovim -u test.vimrc` and this minimal test.vimrc:
```
call plug#begin()
Plug 'jpalardy/vim-slime'
call plug#end()
let g:slime_target = "kitty"
let g:slime_default_config = {"listen_on": $KITTY_LISTEN_ON}
```

Thanks

**Comment by kovidgoyal** [comment]
Works for me running current kitty master with the follwoing steps:

1) kitty --config=NONE --listen-on=unix:/tmp/kitty -o allow_remote_control=socket-only -o enabled_layouts=tall --session <(echo "launch vim setup.py\nlaunch ipython")

1.5) Add the following to ~/.vimrc (and add the vim-slime plugin)

let g:slime_target = "kitty"
let g:slime_bracketed_paste = 1

2) Scroll to some python function and press ctrl-c + ctrl-c in vim
normal mode

3) Press 2 to select the second window (the one in which ipython is
running)

4) Accept the value of KITTY_LISTEN_ON as unix:/tmp/kitty

5) Press ctrl-c crl-c again

6) function is pasted as expected in the right hand side ipython window

**Comment by kovidgoyal** [comment]
![screenshot](https://github.com/kovidgoyal/kitty/assets/1308621/b5ae4fe6-68e3-43d4-a803-7f1a187f6bbe)

Screenshot.

**Comment by kovidgoyal** [comment]
I did the test on linux, as I dont have all this available in a macOS envoronment but I dont really see any reason to expect it to work differently on macOS

**Comment by xxyxxyxyx1** [comment]
Thank you very much for testing this yourself. 

I compiled kitty from master and followed your steps above exactly. I still have the same problem: the slime works correctly, but afterwards, most keyboard shortcuts are broken. I must use the mouse just to quit the kitty window. 

I guess this problem is with either macOS, or something particular to my environment. I will try to reproduce this on a different macOS computer and report back. If you have any other ideas to try please let me know.

**Comment by ad-chaos** [comment]
I can reproduce this with just `kitty @ select-window --self`

1. `kitty --config=NONE --listen-on=unix:/tmp/kitty -o allow_remote_control=socket-only -o enabled_layouts=tall --session <(echo "launch\nlaunch")`
2. `kitty @ select-window --self`
3. select any window
4. Now see that opening a new tab `cmd+t`, a new window `cmd+enter`, or closing kitty `cmd+q` doesn't work.
5. Open vim, enter insert mode, press `shift+3` and `3` is entered, `shift+4` and `4` is entered, try quitting vim with `:q` and observe you can't (because `shift+;` enters `;`).

`shift+letter` keys work on the other hand, and the `shift+num` keys work in nano and on the shell prompt just not in vim.

**Comment by ad-chaos** [comment]
I can also see the following on stdout
```
[027 12:09:52.605641] [PARSE ERROR] Unrecognized DCS code: 0x7a
[027 12:09:52.605656] [PARSE ERROR] Unknown CSI code: 'm' with start_modifier: '' and end_modifier: '%' and parameters: '0'
```

**Comment by xxyxxyxyx1** [comment]
I have recompiled kitty from master version 8c12086be, and it appears the problem has been fixed! Thank you both. (And thank you for making compiling kitty so effortless to a beginner.) 
I will comment again if I find anything is still broken.

**Comment by kovidgoyal** [comment]
It's already released.

**Comment by ed9w2in6** [comment]
Sorry for being ambiguous.

I am referring to version `0.32.2` which I assume will contain commit `5a2ee2f`, when will it be released?

You are referring to the nightly build, which has already been released right?

**Comment by kovidgoyal** [comment]
Just use a nightly. I dont have any fixed plans to release 0.32.2
